src/Domain/Creators/TimeDependence/ScalingAndZRotation.hpp

```diff
@@ -0,0 +1,198 @@
+ * radial scaling can be used by specifying the `UseLinearScaling` bool.
+ */
+template <size_t MeshDim>
+class ScalingAndZRotation final : public TimeDependence<MeshDim> {
+  static_assert(
+      MeshDim > 1,
+      "ScalingAndZRotation<MeshDim> undefined for MeshDim == 1");
+
+ private:
+  using Identity = domain::CoordinateMaps::Identity<1>;
+  using Rotation = domain::CoordinateMaps::TimeDependent::Rotation<2>;
+  using CubicScaleMap =
+      domain::CoordinateMaps::TimeDependent::CubicScale<MeshDim>;
+
+ public:
+  using maps_list = tmpl::list<
+      domain::CoordinateMap<Frame::Grid, Frame::Inertial, Rotation>,
+      domain::CoordinateMap<Frame::Grid, Frame::Inertial, CubicScaleMap>,
+      domain::CoordinateMap<
+          Frame::Grid, Frame::Inertial,
+          CoordinateMaps::TimeDependent::ProductOf2Maps<Rotation, Identity>>>;
+
+  static constexpr size_t mesh_dim = MeshDim;
+
+  /// \brief The initial time of the function of time.
+  struct InitialTime {
+    using type = double;
+    static constexpr Options::String help = {
+        "The initial time of the function of time"};
+  };
+  /// \brief The \f$x\f$-, \f$y\f$-, and \f$z\f$-velocity.
+  struct AngularVelocity {
+    using type = double;
+    static constexpr Options::String help = {
+        "The angular velocity of the map."};
+  };
+
+  /// \brief The outer boundary or pivot point of the
+  /// `domain::CoordinateMaps::TimeDependent::CubicScale` map
+  struct OuterBoundary {
+    using type = double;
+    static constexpr Options::String help = {
+        "Outer boundary or pivot point of the map"};
+  };
+  /// \brief The initial values of the expansion factors.
+  struct InitialExpansion {
+    using type = std::array<double, 2>;
+    static constexpr Options::String help = {
+        "Expansion values at initial time."};
+  };
+  /// \brief The velocity of the expansion factors.
+  struct Velocity {
+    using type = std::array<double, 2>;
+    static constexpr Options::String help = {"The rate of expansion."};
+  };
+  /// \brief The acceleration of the expansion factors.
+  struct Acceleration {
+    using type = std::array<double, 2>;
+    static constexpr Options::String help = {"The acceleration of expansion."};
+  };
+  /// \brief Whether to use linear scaling or cubic scaling.
+  struct UseLinearScaling {
+    using type = bool;
+    static constexpr Options::String help = {
+        "Whether or not to turn on cubic scaling."};
+  };
+
+  using MapForComposition = detail::generate_coordinate_map_t<
+      tmpl::list<CubicScaleMap,
+                 tmpl::conditional_t<MeshDim == 2, Rotation,
+                                     domain::CoordinateMaps::TimeDependent::
+                                         ProductOf2Maps<Rotation, Identity>>>>;
+
+  using options =
+      tmpl::list<InitialTime, AngularVelocity, OuterBoundary, UseLinearScaling,
+                 InitialExpansion, Velocity, Acceleration>;
```

Comment:
Why not take a `CubicScale` and `UniformRotationAboutZAxis` time dependence as options instead (or pointers to those)? All these options are already defined there. You could also use their respective `functions_of_time` functions so that isn't duplicated either.

Reply:
I have mixed feelings about this. Doing this would essentially bring back a good portion of the Composition map in a less general way.

The upside would be fewer duplicated options.  The downside is that I would need a way to get the options from `CubicScale` and `UniformRotationAboutZAxis` into `ScalingAndZRotation`.  To do this, `ScalingAndZRotation` could use the `functions_of_time` and  `grid_to_inertial_map` functions of those classes [basically re-doing what was done in the Composition map].  If I would do this, `grid_to_inertial_map()` would need to stay public. [I wanted to make it private in response to Nils' comment].   And not sure what would happen in my next PR when I change some of the maps to add a Distorted frame (but I could probably make it work).

Thoughts?

Reply:
Ok yeah I don't think I know enough about what you plan to do when adding the distorted frame to be convinced that avoiding code duplication is the best option here. I'm ok with leaving this as is. We just have to remember that if we change `CubicScale` or `UniformRotationAboutZAxis` time-dependence, we'll have to change this as well.

Reply:
Ok, I'll keep it as is for now.  After my next PR we might re-evaluate.